src/Microsoft.AspNet.Buffering/BufferingWriteStream.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.AspNet.Buffering
+{
+    internal class BufferingWriteStream : Stream
+    {
+        private readonly Stream _innerStream;
+        private readonly MemoryStream _buffer = new MemoryStream();
+        private bool _shouldBuffer = true;
```

Comment:
Is there a reason that this uses the term "should"? Is it merely a suggestion, or is it stating factually whether buffering is to be used at this moment? Looking at the code it *seems* to me that it is the latter.